jquery.password_field.js

```diff
@@ -64,6 +152,115 @@
             });
 
         });
-    }
 
-}(jQuery));
\ No newline at end of file
+    };
+
+    /**
+     * Adds a strength indicator under the password field.
+     * The indicator has two parts:
+     * - a strength indicator base on <a href="https://github.com/dropbox/zxcvbn">zxcvbn</a> library.
+     * - a validator based on password rules: > 8 chars, upper and lowercase chars, at lease on digit
+     *
+     * In the options the displayed texts can be controlled.
+     *
+     * Example:
+     * <code><pre>
+     * {
+     *   strength_text: {
+     *     weak: 'Weak',
+     *     okay: 'Okay',
+     *     medium: 'Medium',
+     *     good: 'Good',
+     *     strong: 'Strong',
+     *     na: ''
+     *   },
+     *   validity: {
+     *     header: 'Your password must have',
+     *     pwd_length: '8 or more characters',
+     *     pwd_upper_and_lower: 'Upper & lowercase letters',
+     *     pwd_digits: 'At least one number'
+     *   }
+     * }
+     * </pre></code>
+     */
+    $.fn.strengthIndicator = function(options) {
+
+        var settings = $.extend({
+            strength: {
+                weak: 'Weak',
+                okay: 'Okay',
+                medium: 'Medium',
+                good: 'Good',
+                strong: 'Strong',
+                na: ''
+            },
+            validity: {
+                header: 'Your password must have',
+                pwd_length: '8 or more characters',
```

Comment:
let's not use pwd. We should spell it out. We should also use camelCase in javascript instead of snake_case